packages/js/components/src/experimental-tree-control/select-tree.tsx

```diff
@@ -0,0 +1,193 @@
+import { SelectedItems } from '../experimental-select-control/selected-items';
+import {
+	getItemLabelType,
+	getItemValueType,
+} from '../experimental-select-control/types';
+import {
+	defaultGetItemLabel,
+	defaultGetItemValue,
+} from '../experimental-select-control/utils';
+import { ComboBox } from '../experimental-select-control/combo-box';
+import { SuffixIcon } from '../experimental-select-control/suffix-icon';
+
+interface SelectTreeProps extends Omit< TreeControlProps, 'getItemLabel' > {
+	selected?: Item[];
+	getItemLabel?: getItemLabelType< Item >;
+	getItemValue?: getItemValueType< Item >;
+	getSelectedItemProps?: any;
+	getFilteredItems: (
+		allItems: Item[],
+		inputValue: string,
+		selectedItems: Item[],
+		getItemLabel: getItemLabelType< Item >
+	) => any[];
+	treeRef?: React.ForwardedRef< HTMLOListElement >;
+	suffix?: JSX.Element | null;
+	isLoading?: boolean;
+	label: string | JSX.Element;
+	onInputChange?: ( value: string | undefined ) => void;
+}
+
+export const SelectTree = function SelectTree( {
+	items,
+	getItemLabel = ( item ) => item?.name || '',
+	getItemValue = ( item ) => item?.id || '',
+	getSelectedItemProps,
+	getFilteredItems,
+	treeRef: ref,
+	suffix = <SuffixIcon icon={ search } />,
+
+	placeholder,
+	isLoading,
+	...props
+}: SelectTreeProps ) {
+	const filteredItems = getFilteredItems(
+		items,
+		props.createValue || '',
+		props.selected || [],
+		getItemLabel
+	);
+	const linkedTree = useLinkedTree( filteredItems );
+
+	const [ isFocused, setIsFocused ] = useState( false );
+
+	const comboBoxRef = useRef< HTMLDivElement >( null );
+
+	// getting the parent's parent div width to set the width of the dropdown
+	const comboBoxWidth =
+		comboBoxRef.current?.parentElement?.parentElement?.getBoundingClientRect()
+			.width;
+
+	return (
+		<Dropdown
+			className="woocommerce-experimental-select-tree-control__dropdown"
+			contentClassName="woocommerce-experimental-select-tree-control__dropdown-content"
+			focusOnMount={ false }
+			renderContent={ ( { onClose } ) =>
+				isLoading ? (
+					<div
+						style={ {
+							width: comboBoxWidth,
+						} }
+					>
+						<Spinner />
+					</div>
+				) : (
+					<Tree
+						{ ...props }
+						ref={ ref }
+						items={ linkedTree }
+						onTreeBlur={ onClose }
+						style={ {
+							width: comboBoxWidth,
+						} }
+					/>
+				)
+			}
+			renderToggle={ ( { isOpen, onToggle, onClose } ) => (
+				<div
+					className={ classNames(
+						'woocommerce-experimental-select-control',
+						{
+							'is-focused': isFocused,
+						}
+					) }
+				>
+					<label
+						htmlFor="categories-input"
+						className="woocommerce-experimental-select-control__label"
+					>
+						{ props.label }
+					</label>
+					<ComboBox
+						comboBoxProps={ {
+							className:
+								'woocommerce-experimental-select-control__combo-box-wrapper',
+							ref: comboBoxRef,
+						} }
+						inputProps={ {
+							className:
```

Comment:
This will also need some `aria` labels for combobox in order to support screen readers. Like: `aria-autocomplete="list" aria-controls="downshift-0-menu" aria-labelledby="downshift-0-label" autocomplete="off"`
Might be worth what `downshift` is doing with the experimental select control (that's where the above comes from).

Reply:
The `aria-controls` will have to match an `id` on the tree control wrapper.
See here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role
for more info.

Reply:
Thanks for pointing this out! I was doing AB testing yesterday at the end of the day and I noticed that the screen reader stopped giving some additional information. I will look into it.